edx/cc2olx
# [BD-27] [TNL-7694] [BB-3248] Support for Wiki Content in cc2olx

This PR adds support for processing canvas wiki links that defined by ``$WIKI_REFERENCE$``.

**JIRA tickets**: 
- https://openedx.atlassian.net/browse/TNL-7694
- https://tasks.opencraft.com/browse/BB-3248

~~**Discussions**:~~

**Dependencies**: None

~~**Screenshots**:~~

~~**Sandbox URL**:~~

**Merge deadline**: None

**Testing instructions**:

1. Checkout this PR.
2. Use a [canvas_cc](https://github.com/instructure/canvas_cc) common cartridge ``.imscc`` file that contains wiki pages.
3. Convert it using ``cc2olx -i <file_path>``
5. Create a new course in your local edX devstack.
4. Import output file from ``output/<file_name>.tar.gz`` via studio.
5. Check that links that had ``$WIKI_REFERENCE$`` has been replaced by ``/jump_to_id/<unit_url_name>``
6. Clicking those links from LMS redirects to the correct place.

**Author notes and concerns**:
1. To achieve this, we needed a map that points CC ``webcontent['href'] -> webcontent['identifier']`` named ``resource_id_by_href``.
2. We set OLX ``url_name`` of chapter, sequential, vertical, and unit with their respective ``identifierref`` from CC
3. When processing static URLs we try to find the appropriate unit ``url_name`` from ``resource_id_by_href`` map and replace them using edX ``/jump_to_id/<unit_id>`` pattern.

**Reviewers**

~~**Settings**~~
